annotation/src/main/java/org/commonjava/maven/ext/annotation/ConfigValueProcessor.java

```diff
@@ -0,0 +1,139 @@
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+@SupportedAnnotationTypes( "org.commonjava.maven.ext.annotation.ConfigValue" )
+@SupportedOptions( { "generationDirectory", "packageName", "rootDirectory" } )
+@SupportedSourceVersion( SourceVersion.RELEASE_8)
+public class ConfigValueProcessor extends AbstractProcessor
+{
+    private static final String GENERATION_DIR = "generationDirectory";
+    private static final String PACKAGE_NAME = "packageName";
+    private static final String ROOT_DIR = "rootDirectory";
+
+    private final Map<String,String> varResults = new HashMap<>( );
+    private final Map<String,String> indexResults = new TreeMap<>( );
+
+    @Override
+    public boolean process( Set<? extends TypeElement> annotations, RoundEnvironment roundEnv )
+    {
+        Messager messager = processingEnv.getMessager();
+
+        if ( annotations.size() == 0 )
+        {
+            return false;
+        }
+        else if ( processingEnv.getOptions().size() == 0 )
+        {
+            messager.printMessage( Diagnostic.Kind.OTHER, "No options supplied ; use maven-annotation-plugin to pass options through" );
+            return false;
+        }
+
+        for ( TypeElement typeElement : annotations )
+        {
+            Set<VariableElement> vElements = ElementFilter.fieldsIn( roundEnv.getElementsAnnotatedWith( typeElement ) );
+
+            for (VariableElement vElement : vElements )
+            {
+                ConfigValue annotation = vElement.getAnnotation( ConfigValue.class );
+
+                messager.printMessage ( Diagnostic.Kind.NOTE,
+                                        "Found " + vElement.toString() + " & " + annotation.docIndex() + " & " + vElement.getConstantValue());
+
+                varResults.put( vElement.toString(), vElement.getConstantValue().toString() );
+                indexResults.put( vElement.getConstantValue().toString(), annotation.docIndex() );
+            }
+        }
+
+        if (!varResults.isEmpty())
+        {
+            try
+            {
+                generateCode();
+            }
+            catch ( IOException e )
+            {
+                messager.printMessage( Diagnostic.Kind.ERROR, "Unable to write file: " + e.toString() );
+                throw new RuntimeException( "Unable to write file", e);
+            }
+        }
+
+        return true;
+    }
+
+    void generateCode() throws IOException
+    {
+        ClassName hashMap = ClassName.get( "java.util", "HashMap");
+        ClassName str = ClassName.get( String.class );
+        ParameterizedTypeName mainType = ParameterizedTypeName.get( hashMap, str, str );
+
+        Set<CodeBlock> contents = new HashSet<>(  );
+        varResults.forEach( ( key, value ) -> contents.add(
+                        CodeBlock.builder().addStatement( "allConfigValues.put($S, $S);", key, value ).build() ) );
```

Comment:
The `;` shouldn't be necessary here as the statement already apparently includes one, leading to double `;;`.